tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,104 @@
+- `networks`: will have the Main Net network where the Neo Token contract is.
+
+After tweaking the `cpm.yaml` file it should be something like this:
+
+```yaml
+defaults:
+  contract-source-network: mainnet
+  contract-generate-sdk: true
+  contract-download: false
+  off-chain:
+    languages:
+    - ts
+    destinations:
+	    ts: src/contracts
+
+contracts:
+  - label: NEO token
+    script-hash: '0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5'
+
+networks:
+  - label: mainnet
+    hosts:
+      - 'https://mainnet1.neo.coz.io:443'
+      - 'http://seed1.neo.org:10332'
+```
+
+> For more information on how to configure `cpm.yaml`, check out [CPM's documentation](https://github.com/CityOfZion/cpm/blob/master/docs/config.md)
+
+Then just run `cpm run` on the terminal and the SDK should be generated at the `src/contracts/neotoken` folder.' Anytime the contract adds or removes method signatures, you'll have to run `cpm run` again to update your SDK.
+
+## 3. Initializing and running the SDK
+Now that the SDK is already generated, you'll need to install the dependencies: [`@types/node`](https://www.npmjs.com/package/@types/node)], and  [`@cityofzion/neon-dappkit-types`](https://www.npmjs.com/package/@cityofzion/neon-dappkit-types). Then you'll have to import and initialize the SDK to be able to call it.
+To initialize the SDK, you'll first need to instantiate a class that implements `Neo3-Invoker`, like [`NeonInvoker`](https://github.com/CityOfZion/neon-dappkit/blob/main/packages/neon-dappkit/src/NeonInvoker.ts) or [`WalletConnect SDK`](https://github.com/CityOfZion/wallet-connect-sdk/tree/main). If you plan to allow users to log into their accounts and make them do the invocation on your project, then Wallet Connect is the better option, otherwise, NeonInvoker can work too.  
+```ts
+import { NeoToken } from './contracts/neotoken'
+
+import { NeonInvoker } from "@cityofzion/neon-dappkit"
+import WcSdk from '@cityofzion/wallet-connect-sdk-core'
+import SignClient from '@walletconnect/sign-client'
+
+async function createNeoTokenSDKWithWcSDK(): Promise<NeoToken> {
+	let wcSdk
+	let signClient
+
+	signClient = await SignClient.init({
+        projectId: 'a9ff54e3d56a52230ed8767db4d4a810', // the ID of your project on Wallet Connect website
+        relayUrl: 'wss://relay.walletconnect.com', // we are using walletconnect's official relay server
+        metadata: {
+          name: 'MyApplicationName', // your application name to be displayed on the wallet
+          description: 'My Application description', // description to be shown on the wallet
+          url: 'https://myapplicationdescription.app/', // url to be linked on the wallet
+          icons: ['https://myapplicationdescription.app/myappicon.png'], // icon to be shown on the wallet
+        },
+    })
+    wcSdk = new WcSdk(signClient)
+
+    return new NeoToken({
+		scriptHash: NeoToken.SCRIPT_HASH,
+		invoker: wcSdk
+	})
+}
```

Comment:
I just noticed there is a problem on this example: a new wc instance will be created everytime the user want's to get the SDK. Let's try something more similar to this: https://github.com/CityOfZion/neon-dappkit/blob/main/packages/neon-dappkit/WALLET-CONNECT.md